data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/MergeAllAggregateAction.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This
+ * action will add the unique keys of each smaller Event to the overall
+ * groupState,
+ * and will create a combined Event from the groupState on concludeGroup. If
+ * smaller Events have the same keys, then these keys will be overwritten with
+ * the keys of the
+ * most recently handled Event.
+ * 
+ * @since 1.3
+ */
+@DataPrepperPlugin(name = "merge_all", pluginType = AggregateAction.class, pluginConfigurationType = MergeAllAggregateActionConfig.class)
+public class MergeAllAggregateAction implements AggregateAction {
+    static final String EVENT_TYPE = "event";
+    public final Map<String, String> dataTypeMap;
+
+    @DataPrepperPluginConstructor
+    public MergeAllAggregateAction(final MergeAllAggregateActionConfig mergeAllAggregateActionConfig) {
+        this.dataTypeMap = mergeAllAggregateActionConfig.getDataTypes();
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction#
+     * handleEvent(org.opensearch.dataprepper.model.event.Event,
+     * org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput)
+     */
+    @Override
+    public AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
+        final GroupState groupState = aggregateActionInput.getGroupState();
+        if (groupState.size() == 0) {
+            groupState.putAll(event.toMap());
+        }
+        Set<String> eventKeys = event.toMap().keySet();
+        for (String key : eventKeys) {
+            if (!this.dataTypeMap.containsKey(key)) {
+                continue;
+            }
+            Object value = null;
+            String dataType = this.dataTypeMap.get(key);
+            switch (dataType) {
+                case "integer":
+                    value = event.get(key, Integer.class);
```

Comment:
I think if we removed the `data_types` configuration you could have this instead:

```
value = event.get(key, Object.class);
```

Reply:
Going with this route I believe that this action could be renamed to `append`, and then the configuration option could be named something like `keys_to_append`. So the full configuration would look like this

```
processor:
    - aggregate:
            identification_keys: ["test_key"]
            action:
                append:
                    keys_to_append: ["key1", "key2"]
```

If `keys_to_append` is not provided, the default behavior could be to append every value to the end of the list. The name `merge_all` implies that it is similar to the `put_all` aggregate action (https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/PutAllAggregateAction.java), while based on the code here we are not overwriting values when keys are seen again. We are only appending them to a list

Reply:
I agree. I added this optional field keys_to_append. If it's absent append is the default behaviour for all fields. If it's present only those fields are used for append.

Reply:
What if the same value appears multiple times? Does it make sense to "merge" or "append" all of them or only have unique values? For example, if you have "ip_address" as one of the field and there are only 10 unique ip_addresses while aggregating 1000 events, does it make sense to have all of them? Or could this be another option in the configuration?

Reply:
To make the merge logic simpler I did not make any effort to check for duplicates. We can add another configuration and remove duplicates in the conclude phase. Also, how deep should we go to check for duplicates? A field can be a nested array.

Reply:
Simpler option is to provide an option to make the result as a list or a set. A set automatically removes duplicates. If you are aggregating 1000 records and if there are only two or three values, it could save a lot of memory/storage space.